### ISSUE TITLE
exporting only filtered data

### ISSUE THREAD

**Issue by milosodalovic** [clarketm/TableExport#17]
Hi,
Let's imagine there is a table with some filtering options enabled. If some filters are selected, table is going to reduce its size. I can't find a way to export only remained data, after the filtering is applied?

**Comment by rifqierusyadi** [comment]
you can filtering data with [datatables](https://datatables.net/)
`var t = $('#tableID').DataTable();`
and put input at above your table.
`<input type="text" id="search" class="search" placeholder="Search...">`.
create keyup function for search or filter datatables.
`$('#search').on('keyup', function () {
t.search( this.value ).draw();
e.reset();
    });`
if you get error like this `uncaught ReferenceError: settings is not defined`, open file tableexport.js find `reset: function () {
                return new TableExport(this.selectors, settings, true);
            },` and change with `reset: function () {
                return new TableExport(this.selectors, this.settings, true);
            },`
and done.

**Comment by clarketm** [comment]
Have you tried adding the `tableexport-ignore` class to the table cell, then running `.reset()` on the `TableExport` object? 

> The following will exclude all `<td>` elements with the `tableexport-ignore` class from the exported results.

### e.g.
```js
var table = $("table").tableExport();
// ...
// add the `tableexport-ignore` class to the cells you want to exclude.
// ...
tables.reset()
```

-------------
Alternatively, if you are only filtering rows and columns I suggest using the `ignoreRows` or `ignoreCols` property supplying an array of indices to exclude from the exported content, then run the `.update()` method to reinitialize `TableExport`.

> The following will exclude the **first** and **third** rows from the exported results.

### e.g.
```js
var table = $("table").tableExport();
// ...
// determine the `ignoreRows` indices to exclude.
// ...
tables.update({
    ignoreRows: [0, 2]   
});
```

[refer to the docs](https://www.travismclarke.com/tableexport/)

**Comment by clarketm** [comment]
Any news of whether using `tableexport-ignore`, `ignoreRows`, or `ignoreCols` aided in resolving your issue?

**Comment by milosodalovic** [comment]
Thanks for support!
I had to make this work very quickly when issue had arisen. Therefore, I've found alternative way and abandoned this tool, although I like it.

Thanks again.

On 16 Oct 2016, at 06:14, Travis Clarke <notifications@github.com<mailto:notifications@github.com>> wrote:


Any news of whether using tableexport-ignore, ignoreRows, or ignoreCols aided in resolving your issue?

-

**Comment by clarketm** [comment]
The **reset** is fixed as of [v3.3.9]() thanks to the [PR](https://github.com/clarketm/TableExport/pull/33) from @Alplob
